cabal-install/Distribution/Client/CmdShowBuildInfo.hs

```diff
@@ -0,0 +1,254 @@
+import Distribution.Verbosity
+         ( Verbosity, silent )
+import Distribution.Simple.Utils
+         ( wrapText, die')
+import Distribution.Types.UnitId (UnitId)
+
+import qualified Data.Map as Map
+import qualified Distribution.Simple.Setup as Cabal
+import Distribution.Client.SetupWrapper
+import Distribution.Simple.Program ( defaultProgramDb )
+import qualified Distribution.Client.InstallPlan as InstallPlan
+import Distribution.Client.ProjectPlanning.Types
+import Distribution.Client.ProjectPlanning (
+  setupHsConfigureFlags, setupHsConfigureArgs,
+  setupHsBuildFlags, setupHsBuildArgs, 
+  setupHsScriptOptions
+  )
+import Distribution.Client.DistDirLayout (distBuildDirectory)
+import Distribution.Client.Types ( PackageLocation(..), GenericReadyPackage(..) )
+import Distribution.Client.JobControl (newLock, Lock)
+import Distribution.Simple.Configure (tryGetPersistBuildConfig)
+import Data.List (find)
+
+showBuildInfoCommand :: CommandUI (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags, TestFlags)
+showBuildInfoCommand = Client.installCommand {
+  commandName         = "new-show-build-info",
+  commandSynopsis     = "Show project build information",
+  commandUsage        = usageAlternatives "new-show-build-info" [ "[TARGETS] [FLAGS]" ],
+  commandDescription  = Just $ \_ -> wrapText $
+        "Build one or more targets from within the project. The available "
+     ++ "targets are the packages in the project as well as individual "
+     ++ "components within those packages, including libraries, executables, "
+     ++ "test-suites or benchmarks. Targets can be specified by name or "
+     ++ "location. If no target is specified then the default is to build "
+     ++ "the package in the current directory.\n\n"
+
+     ++ "Dependencies are built or rebuilt as necessary. Additional "
+     ++ "configuration flags can be specified on the command line and these "
+     ++ "extend the project configuration from the 'cabal.project', "
+     ++ "'cabal.project.local' and other files.",
+  commandNotes        = Just $ \pname ->
+        "Examples:\n"
+     ++ "  " ++ pname ++ " new-build\n"
+     ++ "    Build the package in the current directory or all packages in the project\n"
+     ++ "  " ++ pname ++ " new-build pkgname\n"
+     ++ "    Build the package named pkgname in the project\n"
+     ++ "  " ++ pname ++ " new-build ./pkgfoo\n"
+     ++ "    Build the package in the ./pkgfoo directory\n"
+     ++ "  " ++ pname ++ " new-build cname\n"
+     ++ "    Build the component named cname module Distribution.Client.InstallPlanin the project\n"
+     ++ "  " ++ pname ++ " new-build cname --module Distribution.Client.InstallPlanenable-profiling\n"
+     ++ "    Build the component in profilingmodule Distribution.Client.InstallPlan mode (including dependencies as needed)\n\n"
+
+     ++ cmdCommonHelpTextNewBuildBeta
+   }
+
+
+-- | The @build@ command does a lot. It brings the install plan up to date,
```

Comment:
reference to the `@build@` command, should say `@show-build-info@` instead. You should go over the code and fix that everywhere. `commandNotes` above also has this problem.